sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetColumnsOperation.scala

```diff
@@ -126,12 +124,47 @@ private[hive] class SparkGetColumnsOperation(
     HiveThriftServer2.eventManager.onStatementFinish(statementId)
   }
 
+  private def getColumnSize(typ: DataType): Option[Int] = typ match {
+    case StringType | BinaryType => None
+    case ArrayType(et, _) => getColumnSize(et)
+    case MapType(kt, vt, _) =>
+      val kSize = getColumnSize(kt)
+      val vSize = getColumnSize(vt)
+      if (kSize.isEmpty || vSize.isEmpty) {
+        None
+      } else {
+        Some(kSize.get + vSize.get)
+      }
+    case StructType(fields) =>
+      val sizeArr = fields.map(f => getColumnSize(f.dataType))
+      if (sizeArr.contains(None)) {
+        None
+      } else {
+        Some(sizeArr.map(_.get).sum)
+      }
+    case other => Some(other.defaultSize)
+  }
+
+  private def getDecimalDigits(typ: DataType): Option[Int] = typ match {
+    case BooleanType | _: IntegerType => Some(0)
+    case FloatType => Some(7)
+    case DoubleType => Some(15)
```

Comment:
how do we pick these numbers?

Reply:
They are from hive's `TypeDescriptor` and I also checked the results, which are same from hive and spark, e.g. `select 0.12345678901234567890D`

Reply:
the result is `0.12345678901234568` which have 17 decimal digits but hive says its 15 for double...

Reply:
It  seems to follow `IEEE 754` https://en.wikipedia.org/wiki/IEEE_754

Reply:
Can we add a code comment to explain it?

Reply:
OK